decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+between specifications - both across DCCs and shared between the OpenAssetIO community. These types
+of entities, relationships and locales can't be counted on to be exactly the same, but defining the
+similarities between them so that they can fulfill reliable behaviours is important so that the
+specifications have sharable meaning.
+
+OpenAssetIO requires a mechanism capable of precisely describing entities, relationships and
+locales with the following criteria:
+
+Hierarchical or Composable (to support coercion of custom definitions to well-known behavioural
+traits) Simple in structure (to simple data types for interop/serialisation) Robust (compile-time
+checks for programming errors - property access or invalid method arguments) Discoverable (runtime
+introspection/help) Definable out of core (by host integrations/managers)
+
+## Options considered
+
+### Option 1
+
+Hierarchical specifications with a single parent per child. We assume that every child fulfills the
+contract of its parent. For example, that all alembic files are a geometry cache, and that all
+geometry caches might have coordinate system based data, and that all geometry caches are a type of
+binary blob. This could be described in a specification: entity. blob.geometry._cache.alembic.
+Another example could be an exr render, which could be a type of image, which could also be a type
+of blob binary data, represented as: entity.blob.image.exr .
+
+#### Pros
+
+- only one way to represent the specification
+- automatic fall back mechanism through inheritance
+- proven in real-world, working scenarios
+
+#### Cons
+
+- difficult to insert new specifications in between types, requires updating the child type
+  identifier of the new insert
+- requires knowledge of as many possible entities as possible to be efficiently taxonomied into an
+  efficient and useful hierarchy
+- sharing behaviours between taxonomically unrelated types - for example a frame range applied to
+  both a geometry cache and an image - is difficult, it relies on duplicated property definitions
+  on each of specifications, meaning that the trait of “having a frame range becomes” implied
+  though well-known property names, rather than an explicitly stated trait.
+- in order to share well-known base specification hierarchies with the OpenAssetIO
+  community,interop is facilitated through the use of common base specifications. If these base
+  hierarchies change, every copy of the specifications will no longer be able to interact with each
+  other despite possibly having the same expected behaviour
+
+### Option 2
+
+Trait based compositional specifications. Specifications are made from generic traits that
+guarantee certain behaviours or consumption patterns from the Manager or Host. The alembic cache
+could be represented as the following: entity.geometry.cache. The exrs rendered out from the Option
+1 example could also be represented as entity.image.cache as they are a type of image cache.
```

Comment:
[consider]
```suggestion
1 example could also be represented as `entity.image.cache` as they are a type of image cache.
```

Reply:
added code quotes in e7717b77781fa9927a5dd0e23863ca706d113557